mlflow/transformers/__init__.py

```diff
@@ -548,20 +548,19 @@ def save_model(
             default_reqs = sorted(set(inferred_reqs).union(default_reqs))
         else:
             default_reqs = None
+            if duplicates := _find_duplicate_requirements(pip_requirements):
+                _logger.warning(
+                    "Duplicate packages are present within the pip requirements. "
+                    f"Duplicate packages: {duplicates}. Please manually specify "
+                    "the requirements by using the `pip_requirements` argument in "
+                    "order to prevent unexpected installation issues for this model."
```

Comment:
Or if it doesn't complicate the logic much, how about showing what those duplicated packages are resolved to? For example, `duplicated requirements "pandas > 1.0" and "pandas" are resolved to "pandas > 1.0.`. It's less scary for users than saying "unexpected installation:)